### PR TITLE
Add overloads to specify branch for ContentRequest

### LINKED ISSUE #958: Creating file on a specific branch.
In the [github api](https://developer.github.com/v3/repos/contents/#create-a-file) I can see that I should be able to create files on non-master branches by specifying an optional "branch" parameter.

Is there a way to do this in octokit.net?

Thanks!

### PR DESCRIPTION
**Pull request by M-Zuber**
Fixes #958

### CODE REVIEW

File: Octokit/Models/Request/CreateFileRequest.cs [hunk 1]
```diff
@@ -91,6 +116,18 @@ public CreateFileRequest(string message, string content) : base(message)
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="DeleteFileRequest"/> class.
```
Comment (ryangribble) [inline Octokit/Models/Request/CreateFileRequest.cs]: Typo in the xml doc with the cref to the class name

**Comment by ryangribble** [comment]
I noted one typo of a class name in XmlDoc comments

Also could I suggest updating the unit tests to cover off the addition of the new branch parameter, and also adding an integration test to prove it actually creates the file on the specified branch?

**Comment by M-Zuber** [comment]
Thank you for catching the typo! 
I should be able to add unit tests after the weekend. 

Would you test anything in the unit tests other then that the correct url was called? 

The integration test can create, update, and then delete a file on the same branch. Sounds good to you?

**Comment by ryangribble** [comment]
Yeah if you look at the "crud" integration test I'd probably do the same actions as that, but specifying a named branch 

For unit test since the url is the same anyway, you could check that the CreateFileRequest object passed through, has the correct path/message/content/branch present in it.  Eg  have look at ```PassesRequestObject()``` test [here] (https://github.com/octokit/octokit.net/pull/1073/files#diff-1570280d72efc6265b42171826428691)

**Comment by M-Zuber** [comment]
I wasn't able to get the unit tests to work, hopefully the integration test will be enough

**Comment by ryangribble** [comment]
Hey @M-Zuber that looks good... your integration test is passing for me here.

On the topic of unit tests, your change was mostly to a Request object and it doesnt look like there are many unit tests of those things in octokit.net.  

However if you are interested, I did just send you a pull request that implements unit tests on the CreateFile() DeleteFile() and UpdateFile() calls in RepositoryClient, including making sure your new branch field is correctly passed through to the underlying ApiClient.  What ive found with Octokit unit tests for API clients, is we typically check 
- calls the correct URL
- passes through the request object
- checks parameters for nulls and throws exceptions if paramter is required

File: Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs [hunk 2]
```diff
@@ -111,6 +111,53 @@ public class TheGetContentsMethod
...
+                fileSha = contents.First().Sha;
+
+                await fixture.DeleteFile(
+                    repository.Owner.Login,
+                    repository.Name,
+                    "somefile.txt",
+                    new DeleteFileRequest("Deleted file", fileSha, branchName));
+
+                await Assert.ThrowsAsync<NotFoundException>(
+                    async () => await fixture.GetAllContents(repository.Owner.Login, repository.Name, "somefile.txt"));
```
Comment (khellang) [inline Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs]: I don't think this state machine is necessary. `Assert.ThrowsAsync` expects a `Task`, which `GetAllContents` already returns :smile:

Comment (M-Zuber) [inline Octokit.Tests.Integration/Clients/RepositoryContentsClientTests.cs]: Copy paste error :smile:
Will fix when I'm back by a machine.

**Comment by M-Zuber** [comment]
@ryangribble thank you for the PR. I asked a question there.

**Comment by Thwaitesy** [comment]
@ryangribble when do you think this will be merged?

**Comment by ryangribble** [comment]
I don't have commit rights, am sitting on a few pending PR's myself 😀

**Comment by shiftkey** [comment]
@Thwaitesy @ryangribble apologies for the delay, I'm aiming to get through these by the end of the week. Thankfully we just did a big release, so the next one should be a bit easier to manage...

**Comment by shiftkey** [comment]
Code looks good, tests all pass, and it got a pretty great review too...

**Comment by M-Zuber** [comment]
YAY!!  I got a Stevie!! :grinning:

**Pull request merged**